pyproject.toml

```diff
@@ -5,7 +5,7 @@ requires = [
     "meson-python>=0.13.1",
     "meson>=1.2.1,<2",
     "wheel",
-    "Cython~=3.0.5",  # Note: sync with setup.py, environment.yml and asv.conf.json
+    "Cython==3.1.0rc1",  # Note: sync with setup.py, environment.yml and asv.conf.json
```

Comment:
You don't want to do this, no need to force it since 3.1.0 is brand new (users and distros may have reasons not to upgrade, and there may be bugs).

You can use this instead:
```meson
if cy.version().version_compare('>=3.1.0')
  cython_args += ['-Xfreethreading_compatible=True']
```

The way I did it in https://github.com/scipy/scipy/pull/22796 is a bit more concise than in this PR, also with `cython_args:` passing in a single place.

Reply:
Actually you do have the `cy.version().version_compare('>=3.1.0')` already, so just no need to pin Cython. You'll get 3.1.0 anyway by default now.